hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OMThroughputBenchmark.java

```diff
@@ -0,0 +1,483 @@
+    }
+
+    @Override
+    public void prepare() throws Exception {
+      ensureVolumeAndBucketExist(ozoneClient, volumeName, bucketName);
+      setTimer();
+    }
+  }
+
+  private class DeleteFileTask extends ApiTask {
+
+    @Override
+    void runTask(long counter) throws Exception {
+      UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+      OmKeyArgs keyArgs = new Builder()
+          .setBucketName(bucketName)
+          .setVolumeName(volumeName)
+          .setReplicationConfig(new RatisReplicationConfig(factor))
+          .setKeyName(generateObjectName(counter))
+          .setLocationInfoList(new ArrayList<>())
+          .setAcls(OzoneAclUtil.getAclList(ugi.getUserName(),
+              ugi.getGroupNames(), ALL, ALL))
+          .build();
+
+      getTimer().time(() -> {
+        ozoneManagerClient.getFileStatus(keyArgs);
+        ozoneManagerClient.deleteKey(keyArgs);
+        return null;
+      });
+    }
+
+    @Override
+    public void prepare() throws Exception {
+      ensureVolumeAndBucketExist(ozoneClient, volumeName, bucketName);
+      if (!useExisting) {
+        CreateFileTask createFileTask = new CreateFileTask();
+        createFileTask.prepare();
+        runTests(createFileTask::runTask);
+        createFileTask.getTimer().time().close();
+        getMetrics().remove(createFileTask.getClass().getSimpleName());
+        reinit();
+      }
+      setTimer();
+    }
+  }
+
+  private class OpenFileTask extends DeleteFileTask {
+
+    @Override
+    void runTask(long counter) throws Exception {
+      UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+      OmKeyArgs keyArgs = new Builder()
+          .setBucketName(bucketName)
+          .setVolumeName(volumeName)
+          .setReplicationConfig(new RatisReplicationConfig(factor))
+          .setKeyName(generateObjectName(counter))
+          .setLocationInfoList(new ArrayList<>())
+          .setAcls(OzoneAclUtil.getAclList(ugi.getUserName(),
+              ugi.getGroupNames(), ALL, ALL))
+          .build();
+
+      getTimer().time(() -> {
+        ozoneManagerClient.lookupFile(keyArgs);
+        return null;
+      });
+    }
+  }
+
+  private class FileStatusTask extends DeleteFileTask {
+
+    @Override
+    void runTask(long counter) throws Exception {
+      UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+      OmKeyArgs keyArgs = new Builder()
+          .setBucketName(bucketName)
+          .setVolumeName(volumeName)
+          .setReplicationConfig(new RatisReplicationConfig(factor))
+          .setKeyName(generateObjectName(counter))
+          .setLocationInfoList(new ArrayList<>())
+          .setAcls(OzoneAclUtil.getAclList(ugi.getUserName(),
+              ugi.getGroupNames(), ALL, ALL))
+          .build();
+
+      getTimer().time(() -> {
+        ozoneManagerClient.getFileStatus(keyArgs);
+        return null;
+      });
+    }
+  }
+
+  private class RenameFileTask extends DeleteFileTask {
```

Comment:
What is the difference between `RenameFile` and `RenameKey`?

Reply:
> I think we should:
> 
> 1. Improve existing code: some `BaseFreonGenerator` subclasses have duplicated logic (e.g. OM-based tests), these can be extracted into intermediate level classes.
> 2. Implement missing specific tests using existing/improved plumbing code.
> 
> These can be done in more manageable chunks (one/few "generator" per Jira issue/PR).

The starting point of this tool class is mainly for om performance and stability test, which could independently work without SCM and DN, like what SCMThroughputBenchmark does. 
The suggestions are reasonable, but diverse parts of tests are not as intuitional as integrated NNThroughputBenchmark. This is just a style difference, we could endure it.

Thx @adoroszlai for the comment. 
The PR may not be updated fast. I will try to follow your suggestions.